capa/features/extractors/__init__.py

```diff
@@ -76,6 +76,41 @@ def get_functions(self):
         """
         raise NotImplemented
 
+    def is_library_function(self, va):
+        """
+        is the given address a library function?
+        the backend may implement its own function matching algorithm, or none at all.
+        we accept a VA here, rather than function object, to handle addresses identified in instructions.
+
+        this information is used to:
+          - filter out matches in library functions (by default), and
+          - recognize when to fetch symbol names for called (non-API) functions
+
+        args:
+          va (int): the virtual address of a function.
+
+        returns:
+          bool: True if the given address is the start of a library function.
+        """
+        return False
```

Comment:
note that this is the default implementation in the super class, so things like SMDA don't have to change. subclasses, like viv, can opt in to additional logic here.